internal/tokencache/access_token_cache_item_test.go

```diff
@@ -69,33 +71,34 @@ func TestAccessTokenPopulateFromJSONMap(t *testing.T) {
 		"cached_at":       "100",
 	}
 	testCachedAt := "100"
-	expectedAccessToken := &accessTokenCacheItem{
-		HomeAccountID:    &testHID,
-		Environment:      &env,
-		CachedAt:         &testCachedAt,
+	want := &accessTokenCacheItem{
+		HomeAccountID:    testHID,
+		Environment:      env,
+		CachedAt:         testCachedAt,
 		additionalFields: map[string]interface{}{"extra": "this_is_extra"},
 	}
-	actualAccessToken := &accessTokenCacheItem{}
-	err := actualAccessToken.populateFromJSONMap(jsonMap)
+	got := &accessTokenCacheItem{}
+	err := got.populateFromJSONMap(jsonMap)
 	if err != nil {
 		t.Errorf("Error is supposed to be nil, but it is %v", err)
 	}
-	if !reflect.DeepEqual(actualAccessToken, expectedAccessToken) {
-		t.Errorf("Actual access token %+v differs from expected access token %+v", actualAccessToken, expectedAccessToken)
+	if diff := (&pretty.Config{IncludeUnexported: false}).Compare(want, got); diff != "" {
+		t.Errorf("TestAccessTokenPopulateFromJSONMap(access tokens): -want/+got:\n %s", diff)
 	}
-	actualHID := *actualAccessToken.HomeAccountID
-	if !reflect.DeepEqual(actualHID, testHID) {
-		t.Errorf("Expected home account ID %s differs from actual home account ID %s", actualHID, testHID)
+
+	gotExtra := got.additionalFields["extra"].(string)
+	if gotExtra != "this_is_extra" {
+		t.Errorf("TestAccessTokenPopulateFromJSONMap(extra field): got %s, want %s", gotExtra, "this_is_extra")
 	}
```

Comment:
is this standard go unit test structure? i assume it's in the go docs...haven't seen the want/got before. pretty cool

Reply:
+1 to this, very cool

Reply:
This is closer to standard go testing structure.  The got, want thing is a google internal standard, where got always proceeds want EXCEPT when it its a diff, which is -want/+got.  

Table driven tests are the standard in Go, though a lot of this is single value tests so they might not fit into it as well.  
Cheney has a good write up: https://dave.cheney.net/2019/05/07/prefer-table-driven-tests